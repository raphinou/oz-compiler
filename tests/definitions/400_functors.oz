% Check prepare is executed when functor is applied
local
   F=functor
      export
         echo:Echo
      import
         System(show:ImportedShow)
      prepare
         Test=1000
      define
         {ImportedShow 'define::start'}
         proc {Echo S}
            {PrivateEcho S}
         end
         proc {PrivateEcho S}
            {ImportedShow Test}
            {ImportedShow S}
         end
         {ImportedShow 'define::end'}
   end
   M
in
skip
   M={Module.apply [F]}.1
end

