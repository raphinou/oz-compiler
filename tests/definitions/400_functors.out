'prepare'
'define::start'
'define::end'
